# первое уравнение x = 4  Результат: 2.6457513110645903
# второе уравнение x = 4  Результат: 2.6457513110645903

# первое уравнение x = 5  Результат: 2.0
# второе уравнение x = 5  Результат: 2.0

# первое уравнение x = 6  Результат: 1.7320508075688772
# второе уравнение x = 6  Результат: 1.7320508075688772

# первое уравнение x = 7  Результат: 1.5811388300841898
# второе уравнение x = 7  Результат: 1.5811388300841898

# первое уравнение x = 8  Результат: 1.4832396974191329
# второе уравнение x = 8  Результат: 1.4832396974191324

from math import cos, sin

def func1():
    """
    Функция для вычисления первого уравнения.

    Запрашивает значения переменных x и y от пользователя, 
    вычисляет числитель уравнения и возвращает его, 
    если он неотрицателен.
    """
    print("Введите значения переменных x и y для первого уравнения.")
    while True:  # Бесконечный цикл для получения корректного ввода
        input_values = input("Введенное значение: ")  # Запрашиваем ввод от пользователя
        if input_values.strip() == "":  # Проверяем, не пустой ли ввод
            print("Вы ввели пустое значение, перейдем ко второму уравнению.")  # Сообщаем пользователю об ошибке
            return func2()  # Возвращаемся к началу цикла
        
        try:
            x , y = map(int, input_values.split())  # Преобразуем введенное значение в число с плавающей точкой
            # Вычисляем числитель первого уравнения
            numerator = cos(x)**4 + sin(y)**2 + 0.25 * (sin(2*x))**2 - 1
            
            # Проверяем, что числитель неотрицателен
            if numerator < 0:
                print("Числитель меньше нуля. Попробуйте снова.")
                continue
            
            return numerator  # Возвращаем результат
        except ValueError:
            print("Некорректный ввод. Введите 2 числа через пробел.")

def func2():
    """
    Функция для вычисления второго уравнения.

    Запрашивает значения переменных x и y от пользователя,
    вычисляет числитель уравнения и возвращает его,
    если он неотрицателен и x больше нуля.
    """
    print("Введите значения переменных x и y для второго уравнения.")
    while True:  # Бесконечный цикл для получения корректного ввода
        input_values1 = input("Введенное значение: ")  # Запрашиваем ввод от пользователя
        if input_values1.strip() == "":  # Проверяем, не пустой ли ввод
            print("Вы ввели пустое значение.")  # Сообщаем пользователю об ошибке
            return None  # Возвращаемся к началу цикла
        
        try:
            x , y = map(int, input_values1.split())  # Преобразуем введенное значение в число с плавающей точкой
            if x < 0:  # Проверяем, что x больше нуля
                print("Значение x должно быть больше нуля .")
                return None
            
            # Вычисляем числитель второго уравнения
            numerator1 = sin(y + x) * sin(y - x)
            
            # Проверяем, что числитель неотрицателен
            if numerator1 < 0:
                print("Числитель меньше нуля. Попробуйте снова.")
                continue
    
            return numerator1  # Возвращаем результат
        except ValueError:
            print("Некорректный ввод. Введите 2 числа через пробел.")

# Основной цикл программы
result = func1()  # Вызываем первую функцию и сохраняем результат
if result is not None:  # Проверяем, не является ли результат None
    print(f"Результат первого уравнения: {result}")  # Выводим результат первого уравнения

result1 = func2()  # Вызываем вторую функцию и сохраняем результат
if result1 is not None:  # Проверяем, не является ли результат None
    print(f"Результат второго уравнения: {result1}")  # Выводим результат второго уравнения
else:
    print("Завершение программы.")  # Если результат None, сообщаем о завершении программы
